lblod/frontend-gelinkt-notuleren
# GN-4748: basic support for fetching dynamic decision templates

### Overview
This PR adds basic support for fetching dynamic decision templates from the regulatory statement endpoint.
Specifically it includes the following changes/additions:

**A refactor of the `RegulatoryStatementsFetcher` service**
- Renamed to `TemplateFetcher` to better reflect its purpose
- Add support for fetching templates of different types (not only regulatory statement templates)
- Add support for querying the `contexts`/`disabledInContexts` properties of templates (important for decisions/agendapoints)

**Modifications to models of agendapoint routes**
- Modified `new` route of agendapoints to also fetch decision templates from regulatory statement endpoint
- Modified `edit` route of agendapoints to also fetch decision templates from regulatory statement endpoint

I made the following initial assumptions in this PR:
- The type of a decision template is `http://data.lblod.info/vocabularies/gelinktnotuleren/BesluitTemplate`
- The predicates used to specify template contexts are: `ext:context` and `ext:disabledInContext`


@abeforgit section:

** Modifications to support new structure node, and the removal of the decision nodes in version 22 of the plugins
** bumps and config changes to support v10 editor and v22 plugins
- merge of @piemonkey 's PR to get the deps to be friends
- changed the builder image cause our ember images don't have a node v20 version

##### connected issues and PRs:
[GN-4906](https://binnenland.atlassian.net/browse/GN-4906)
[GN-4748](https://binnenland.atlassian.net/browse/GN-4748?atlOrigin=eyJpIjoiMWM0ZjhiYzU0ZjZmNDYyYmIwNGIzZDM4NDM5YzdlNWIiLCJwIjoiaiJ9)
https://github.com/lblod/frontend-reglementaire-bijlage/pull/236
https://github.com/lblod/app-reglementaire-bijlage/pull/77
https://github.com/lblod/reglement-publish-service/pull/14
Solves https://github.com/lblod/frontend-reglementaire-bijlage/issues/227


### Setup
Check-out https://github.com/lblod/app-reglementaire-bijlage/pull/77 for a description on how to set-up the stack

### How to test/reproduce
Check-out https://github.com/lblod/app-reglementaire-bijlage/pull/77 for a description on how to test the new features.

### Checks PR readiness
